src/core/net/udp6.cpp

```diff
@@ -205,6 +221,35 @@ otError Udp::Bind(SocketHandle &aSocket, const SockAddr &aSockAddr)
     return error;
 }
 
+#if OPENTHREAD_FTD && OPENTHREAD_CONFIG_BACKBONE_ROUTER_ENABLE
+void Udp::BindToNetif(SocketHandle &aSocket, otNetifIdentifier aNetifIdentifier)
+{
+    if (aNetifIdentifier == OT_NETIF_BACKBONE)
+    {
+        SetBackboneSocket(aSocket);
+    }
+}
+
+void Udp::SetBackboneSocket(SocketHandle &aSocket)
+{
+    RemoveSocket(aSocket);
+
+    if (mPrevBackboneSockets != nullptr)
+    {
+        mSockets.PushAfter(aSocket, *mPrevBackboneSockets);
+    }
+    else
+    {
+        IgnoreError(mSockets.Add(aSocket));
+    }
+}
```

Comment:
I think it may be actually much easier and a lot cleaner to use two linked-lists, one as before and one for `BackboneSocket`. This makes all the logic in the core for this much simpler.

I think you mentioned the only limitation is with public `otUdpGetSockets()` behavior.

How about adding a new public API that gets the list of backbone sockets in addition to `otUdpGetSockets()` which would not include the backbone sockets.

We can also later change the public API to an iterator based API to be more flexible.

Reply:
The segmented linked list just works. It's largest advantage is that it does not affect the public API at all.
I would prefer to switch to use multiple linked lists of sockets when it's really required.

Thoughts? @abtink @bukepo

Reply:
It seems we are using the segmented list for managing addresses. I like its simplicity for implementation and also allows providing simpler public APIs(introduce another iterator).

Reply:
Using segmented lists sounds fine to me.